### ISSUE TITLE
Use custom selected location for all files

### ISSUE THREAD

**Issue by mhutti1** [kiwix/kiwix-android#837]
**Is your feature request related to a problem? Please describe.**
See #729 

**Describe the solution you'd like**
All android files we save should be saved to the selected location.

**Comment by Aditya-Sood** [comment]
Went through #729 
TL;DR - No clear upside of introducing this feature. Possibly better to keep things as is.

The Android/data/<package-name>/files directory is deleted upon uninstalling the app, so storing files there goes back to the debate in #596 (whether the ZIM files should stay or not)

For adding custom storage, I think most Kiwix users wouldn't be plugging in their devices or SD cards to computers or be using a file manager app to interact with the structure of the storage, so there might not be any need of such a feature.

Also, since this directory isn't removed upon app removal as of now, allowing the users to store in custom locations can cause confusion later on (about the location) when they wish to remove/share the orphaned ZIM files after Kiwix has been uninstalled. And as the ZIM files are of potential use to entities other than the Kiwix app, it shouldn't qualify as "cluttering" the root.

**Comment by metal450** [comment]
I strongly disagree.

>>The Android/data//files directory is deleted upon uninstalling the app, so storing files there goes back to the debate in #596 (whether the ZIM files should stay or not)

Isn't that the point of having options? Those who are happy with the default behavior can leave the default settings.  Those who prefer the other behavior wouldn't be forced into this behavior, & can electively configure it as they prefer for their devices. It doesn't have to be one or the other/one-size-fits-all.  That's why we have options

>> I think most Kiwix users wouldn't be plugging in their devices or SD cards to computers or be using a file manager app to interact with the structure of the storage

I really don't see a basis for this assumption.  Every Android device I've ever used comes with a built-in file manager (for good reason) - and even with that already being on the system, ES File Explorer alone has 100 million downloads on the Play Store.  And that's just one of many file managers.  Xiaomi's has 10 million, Total Commander has 10 million, etc - clearly many people *are* interacting with their storage in a way that's meaningful enough to seek out 3rd party apps to facilitate doing so.

Likewise, plugging a phone into the computer is extremely common practice for things like copying MP3s onto it or moving photos off of it.  I can tell you that I have explicitly had a non-tech-saavy family member tell me "man, I wish the files on this phone weren't such a mess" while doing so on at least one occasion (completely unprompted by me).

>>allowing the users to store in custom locations can cause confusion later on

If a user takes the effort of explicitly configuring a custom location, they probably aren't the type to be confused by the files appearing in that location. Users who are non tech-saavy enough to be confused by something like this are users who would in general not dig around in & change settings anyway.  But if you're really concerned about this, a much better solution would be to just gate the option behind a warning.  i.e. "Warning: if you change this setting, Kiwix's ZIM files will remain in this location, even if you uninstall this app.  For consistency & predictability, it's suggested that you leave the default option.  Are you sure you want to change this?".  Certainly that's much better than saying "one or two people might configure an option, and then be confused when it behaves according to their configuration, therefore nobody should be able to have that option at all." That sounds more like the Apple approach...but some of us are on Android for a reason ;)

>>And as the ZIM files are of potential use to entities other than the Kiwix app, it shouldn't qualify as "cluttering" the root.

ZIM files can be useful to other entities without having to be in a hard-coded, non-configurable location at the top of the storage.  They could just as easily be retained & used by other entities even if you chose to store them in, say, Storage/CustomAppData/Kiwix, or Storage/MyDocs/Kiwix, or any other system of organization that a particular user may prefer on their own device.

Even setting all of the above aside, I can tell you that I have dozens of apps on my phone, and although there are admittedly a few that behave this way (DJI Go, Maps.Me, WhatsApp), they are in the *strong* minority.  The vast majority of apps that do store data outside of Android/data//files let the user decide where this will be.  There is a reason for this: because it's not useless to let users configure the organization of their own devices.

**Comment by mhutti1** [comment]
Custom locations are now removed as of #1165 so I am closing this.

**Comment by metal450** [comment]
Does that mean it will now use the proper location in Android/data/(app-path)/files?  Or it will just go back to cluttering up the top-level storage in a hard-coded path (per my comment above from Nov 2018)?

**Comment by mhutti1** [comment]
We will continue using our current setup for now. This can vary based on device configuration. If you have a specific issue with this I suggest making a new issue.

**Comment by metal450** [comment]
...So before I do that, just to clarify: what will now happen if an existing install (that had a custom storage location) is updated to the current version?  Will it go back to recreating the hardcoded /Kiwix folder at the root of the storage, and move the files back there? Or will it still keep using them at the previous location?

I just updated to 2.5.2, but in that version at least, the option seems to remain & it doesn't seem to have yet regressed to cluttering up the root-level storage again.  If (even withou the option) it still picks up ZIMs located elsewhere, & doesn't keep recreating that folder, it seems not to be an issue.  *If*.
